.github/workflows/deploy-base.yaml

```diff
@@ -0,0 +1,101 @@
+# base for all deployments of lamp pipeline
+#
+# pull the contents of a branch
+# create a docker image for the lamp_py project
+# push that image to ecr
+# deploy it (optionally) to ingestion, rail performance manager, and tableau publisher
+# report everything to slack
+on:
+  workflow_call:
+    inputs:
+      env-name:
+        description: One of 'prod', 'staging', or 'dev'
+        required: true
+        type: string
+      deploy-ingestion:
+        description: Should the Ingestion Application be Deployed
+        required: false
+        default: false
+        type: boolean
+      deploy-rail-pm:
+        description: Should the Rail Performance Manager Application be Deployed
+        required: false
+        default: false
+        type: boolean
+      deploy-tableau-publisher:
+        description: Should the Tableau Publisher Application be Deployed
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      DOCKER_REPO:
+        description: ECR Docker repo to push to
+        required: true
+      SLACK_WEBHOOK:
+        description: Slack URL to post to
+        required: true
+      ROLE_TO_ASSUME:
+        description: AWS_ROLE_ARN
+        required: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout Branch
+        uses: actions/checkout@v3
+
+      - name: Configure AWS credentials
+        id: setup-aws
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ inputs.role-to-assume }}
+          aws-region: us-east-1
+
+      - name: Build and Push Docker Image
+        id: build-push
+        uses: mbta/actions/build-push-ecr@v2
+        with:
+          role-to-assume: ${{ inputs.role-to-assume }}
+          docker-repo: ${{ inputs.docker-repo }}
+          dockerfile-path: ./python_src/
+
+      - name: Deploy Ingestion Application
+        id: deploy-ingestion
+        if: ${{ inputs.deploy-ingestion }}
+        uses: mbta/actions/deploy-ecs@v2
+        with:
+          role-to-assume: ${{ inputs.role-to-assume }}
+          ecs-cluster: lamp
+          ecs-service: lamp-ingestion-${{ inputs.env-name }}
+          docker-tag: ${{ steps.build-push.outputs.docker-tag }}
+
+      - name: Deploy Rail Performance Manager Application
+        id: deploy-rail-performance-manager
+        if: ${{ inputs.deploy-rail-pm }}
+        uses: mbta/actions/deploy-ecs@v2
+        with:
+          role-to-assume: ${{ inputs.role-to-assume }}
+          ecs-cluster: lamp
+          ecs-service: lamp-rail-performance-manager-${{ inputs.env-name }}
+          docker-tag: ${{ steps.build-push.outputs.docker-tag }}
+
+      - name: Deploy Tableau Publisher Application
+        id: deploy-tableau-publisher
+        if: ${{ inputs.deploy-rail-pm && inputs.env-name == 'prod' }}
+        uses: mbta/actions/deploy-scheduled-ecs@v2
```

Comment:
new mbta action that is based off of the action we used in the dmap-import project.